scripts/build_powerplants.py

```diff
@@ -18,6 +18,31 @@ def country_alpha_2(name):
         cntry = pyc.countries.get(official_name=name)
     return cntry.alpha_2
 
+def add_my_carriers(ppl):
+    switch = snakemake.config['electricity']['my_carriers_switch']
+    if switch == 'replace-all' or switch == 'replace-selection' or switch == 'add':
+        countries_dict = snakemake.config['countries_dict'] # dictionary, eg. GB: United Kindgom
+        for country in snakemake.config['electricity']['my_carriers_for_countries']:
+            dirname = os.path.abspath(os.path.join(os.path.dirname(__file__),".."))
+            add_ppls = pd.read_csv(dirname + "/resources/powerplants_" + country + ".csv", index_col=0)
+            if switch == 'replace-all' or switch == 'replace-selection':
+                to_drop = ppl[ppl.Country == countries_dict[country]]
+                if switch == 'replace-selection':
+                    to_drop = to_drop[to_drop.Fueltype.isin(add_ppls.groupby('Fueltype').mean().index)]
+                ppl = ppl.drop(to_drop.index)
+            ppl = ppl.append(add_ppls, sort='False')
+    else:
+        logger.warning('my_carriers_switch is invalid keyword, try one of [add, replace-all, replace-selection]. powerplants remain unchanged.')
+    return ppl
+
+def restrict_buildyear(ppl):
+    year = snakemake.config['electricity']['restrict_buildyear']
+    search_pattern = [str(int(year)+x) for x in range(1,2050-int(year))]
```

Comment:
Took me a while to reason about, what you are searching for, instead use:
```python
search_pattern = [str(x) for x in range(int(year),2050)]
```
or even
```python
search_pattern = map(str, range(int(year), 2050))
```